xmake/modules/target/action/install/main.lua

```diff
@@ -233,7 +243,8 @@ function _install_moduleonly(target, opt)
 end
 
 function main(target, opt)
-    local installdir = target:installdir()
+    opt = opt or {}
+    local installdir = opt.installdir and target:set("installdir", opt.installdir) or target:installdir()
```

Comment:
we should not call `target:set` to modify target installdir configuration.

Reply:
I was wondering if I could save the installdir of a target, then set it to the old value at the end of the function.
The other solution that comes to mind is to add an opt argument to installdir, libdir and bindir of target.

Which solution do you prefer?

Reply:
why not use `local installdir = opt.installdir and opt.installdir or target:installdir()`?

Reply:
Because when `target:installdir(opt.installdir)` is called, installdir is not set in the target (or not the right value). One solution would be to join installdir and bindir each time, but as this is already what bindir does, I haven't considered this option.

Reply:
you can add `function _get_installdir(target, opt)` to wrap it. then get them each time.

Reply:
like this 

https://github.com/xmake-io/xmake/blob/b9385560675a37623b2d4d385b6290b87b824d66/xmake/plugins/pack/batchcmds.lua#L28-L62